docs/chapter3-payload-handoff-state.rst

```diff
@@ -0,0 +1,164 @@
+possible a similar approach is followed for each architecture.
+
+
+3.2 Requirements for all architectures
+--------------------------------------
+
+* Interrupts must be disabled so that an interrupt during handover will not
+  cause ambiguity as to whether Platform Init or Payload will service the
+  interrupt
+* DMA must be disabled, so as to avoid undefined behaviour.
+* At least 16KB must be available on the stack as passed through to the
+  Payload. The stack must be aligned as defined by the architecture. Payload
+  can set up its own stack from available memory when necessary, so the
+  Platform Init stack should only be used for initial setup.
+
+Some general guidelines:
+
+* Set as few things as possible, because across the range of architectures,
+  and implementations, very little is constant
+* If page tables are set up, try to use GiB pages and coarse mappings. Kernels
+  will change them anyway. It is not essential that we exactly match the page
+  tables to available memory; e.g., we might have 4.5 GiB of memory and set up
+  page tables for 5 GiB. That is acceptable as long as we accurately report
+  memory size to the payload.
+* Correct memory maps are critical. E820 is not sufficient to describe memory
+  maps across all architectures. 
+* Define what parameters need to be passed to the first payload, e.g. a pointer
+  to the root of a table as the first arg, etc. The number of parameters should
+  be kept small.
+
+
+3.3 Architecture
+----------------
+
+This describes the handoff register settings for each architecture:
+
+3.3.1 ARM
+~~~~~~~~~
+
+Register calling:-
+
+32-bit::
+
+   r0: Reserved, must be zero
+   r1: divided into the following fields:
+      r1[23:0]: set to the TL signature (0x6e_d0ff)
+      r1[31:24]: version of the register convention used. Set to 0 for the convention specified in this document.
+   r2: Pointer to FDT
+   r3: Reserved, must be zero
+   r13: Stack pointer
+   r14: Link register (for returning to Platform Init)
+
+64-bit::
+
+   x0: Pointer to FDT
+   x1:  divided into the following fields:
+      x1[23:0]: set to the TL signature (0x6e_d0ff)
+      x1[31:24]: version of the register convention used. Set to 0 for the convention specified in this document.
+      x1[63:32]: reserved, must be zero.
+   x2: Reserved, must be zero
+   x3: Reserved, must be zero
+   sp: Stack pointer
+   r14: Return value, for returning to Platform Init
+
+Additionally:
+
+* Unaligned access must be enabled.
+* If MMU is enabled, MMU configuration must use only 4k pages and a single
+  translation base register (TTBR0)
+
+
+3.3.2 RISC-V
+~~~~~~~~~~~~
+
+We should follow this standard: https://github.com/riscv-software-src/opensbi/blob/master/docs/firmware/fw.md
+
+::
+
+   a0/x10: hartid
+   a1/x11: device tree blob address in memory via a1 register. The address must be aligned to 8 bytes.
+   Stack: The stack is 16-byte aligned and 16kb reserved as mentioned in common section
```

Comment:
- Remove "Stack: The stack is 16-byte aligned and 16kb reserved as mentioned in common section" It seems repetition of what is mentioned in the common section.
- Instead just add "Valid sp containing valid sp address"

Reply:
Add: MMU should be disabled
Remove **word Length** related note.
Remove "A0 contains pointer to hartid ; A1 contains pointer to FDT; all other registers
are poisoned or unset (poisoned is preferred)."
note on poisoning of registers- not required as long as UPL assumes those registers do not contain any valid info.

Reply:
sp == stack pointer register in RV